src/ride/ride.c

```diff
@@ -156,9 +156,29 @@ static void ride_update_vehicle_colours(int rideIndex);
 static void ride_set_vehicle_colours_to_random_preset(rct_ride *ride, uint8 preset_index);
 static void maze_entrance_hedge_removal(int x, int y, rct_map_element *mapElement);
 
+rct_ride *get_ride(int index)
+{
+	if (index < 0 || index >= MAX_RIDES)
+	{
+		log_error("invalid index %d for ride", index);
+		return NULL;
+	}
+	return &g_ride_list[index];
+}
+
+rct_ride_type *get_ride_entry(int index)
+{
+	return gRideTypeList[index];
+}
+
+rct_ride_measurement *get_ride_measurement(int index)
+{
+	return &(RCT2_ADDRESS(RCT2_ADDRESS_RIDE_MEASUREMENTS, rct_ride_measurement)[index]);
+}
+
 rct_ride_type *ride_get_entry(rct_ride *ride)
 {
-	return GET_RIDE_ENTRY(ride->subtype);
+	return gRideTypeList[ride->subtype];
```

Comment:
You may as well just make this call the other function then you don't need to copy any max/min checks.